server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -262,31 +273,43 @@ class DeltaSharingService(serverConfig: ServerConfig) {
       @Param("share") share: String,
       @Param("schema") schema: String,
       @Param("table") table: String,
-      queryTableRequest: QueryTableRequest): HttpResponse = processRequest {
-    if (queryTableRequest.version.isDefined && queryTableRequest.version.get < 0) {
+      request: QueryTableRequest): HttpResponse = processRequest {
+    val numVersionParams = Seq(request.version, request.timestamp, request.startingVersion)
+      .filter(_.isDefined).size
+    if (numVersionParams >= 2) {
+      throw new DeltaSharingIllegalArgumentException(ErrorStrings.multipleParametersSetErrorMsg(
+        Seq("version", "timestamp", "startingVersion"))
+      )
+    }
+    if (request.version.isDefined && request.version.get < 0) {
       throw new DeltaSharingIllegalArgumentException("table version cannot be negative.")
     }
+    if (request.startingVersion.isDefined && request.startingVersion.get < 0) {
+      throw new DeltaSharingIllegalArgumentException("startingVersion cannot be negative.")
+    }
 
     val start = System.currentTimeMillis
     val tableConfig = sharedTableManager.getTable(share, schema, table)
-    if (queryTableRequest.version.isDefined || queryTableRequest.timestamp.isDefined) {
+    if (numVersionParams > 0) {
       if (!tableConfig.cdfEnabled) {
         throw new DeltaSharingIllegalArgumentException("Reading table by version or timestamp is" +
           " not supported because change data feed is not enabled on table: " +
           s"$share.$schema.$table")
```

Comment:
Do we need to change this? Will streaming will be permitted on non-cdf-enabled tables?

Reply:
Streaming will be permitted on non-cdf-enabled tables, still need some guard.
And we'll changes this once we finalize on the default behavior change, since with that decision, the guard field  will change. It's sort of orthogonal to the streaming functionality, so will change it in a separate PR.

Reply:
Consider adding a todo for now?